nemo/collections/tts/models/fastpitch.py

```diff
@@ -142,16 +146,23 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         speaker_emb_condition_prosody = cfg.get("speaker_emb_condition_prosody", False)
         speaker_emb_condition_decoder = cfg.get("speaker_emb_condition_decoder", False)
         speaker_emb_condition_aligner = cfg.get("speaker_emb_condition_aligner", False)
+        energy_embedding_kernel_size = 0
+        energy_predictor = None
+        if self._cfg.get("energy_predictor", None) is not None:
+            energy_predictor = instantiate(self._cfg.energy_predictor)
+            energy_embedding_kernel_size = cfg.get("energy_embedding_kernel_size", 0)
```

Comment:
initializing `energy_embedding_kernel_size` and `energy_predictor` at line 149 and 150 is not necessary, because `self._cgf.get()` already take care of the default values if they do not exist. 

This also brings in a question: shall we move the `if...else..` to the `__init__` of `FastPitchModule` for the case when `energy_embedding_kernel_size` and `energy_predictor` do not exist? It is because you have to pass them to `FastPitchModule` no matter what values they are.

Reply:
You are right about line 149. We do not need to initialize `energy_embedding_kernel_size` the `cfg.get` will take care of that. But `energy_predictor` needs to be initialized to None. Because `energy_predictor` in line 150 is variable for the Neural module for predicting energy. Whereas, `self._cfg.get("energy_predictor", None)` will give us just the config, it is only when we call `instantiate` in line 152 that `energy_predictor` gets it's value (the neural module).

Can we move the `if...else`: I don't think so. Because `energy_predictor` is the NeuralModule that we pass to FastPitchModule, what we get from cfg.get is just the config. And for consistency all initialization of Neuralmodules is done in the FastPitch init and not in FastPitchModule init.

Reply:
> energy_predictor needs to be initialized to None

It is still not necessary because if the input cfg is `None` then `instantiate` func will directly return `None` (ref: https://github.com/facebookresearch/hydra/blob/main/hydra/_internal/instantiate/_instantiate2.py#L178-L179). So in this case, you only need,

```python
energy_predictor = instantiate(self._cfg.get("energy_predictor", None))
```